packages/contracts-sdk/src/lib/contracts-sdk.ts

```diff
@@ -941,13 +944,24 @@ export class LitContracts {
     });
 
     const networks = activeValidatorStructs.map((item: ValidatorStruct) => {
-      const protocol = HTTP_BY_NETWORK[network];
       const centralisation = CENTRALISATION_BY_NETWORK[network];
-      const ip = intToIP(item.ip);
-      const port = item.port;
 
+      // Convert the integer IP to a string format
+      const ip = intToIP(item.ip);
+      let port = item.port;
+
+      // Determine the protocol to use based on various conditions
+      const protocol =
+        // If nodeProtocol is defined, use it
+        nodeProtocol ||
+        // If port is 443, use HTTPS, otherwise use network-specific HTTP
+        (port === 443 ? HTTPS : HTTP_BY_NETWORK[network]) ||
+        // Fallback to HTTP if no other conditions are met
+        HTTP;
```

Comment:
There is no port+protocol check? Is it valid that a validator uses https on port 80 or http on port 443?
Maybe forcing https on port 443 to at least avoid that

Reply:
It’s a good point. AFAIK. a validator might use `https` on port `80`, and I actually wanted to force `https` on port `443`. However, I think we should still provide an option to override the default value using nodeProtocol (from an internal developer’s perspective).